core/src/main/scala/io/finch/Endpoints.scala

```diff
@@ -403,8 +404,9 @@ trait Endpoints {
   val binaryBodyOption: Endpoint[Option[Array[Byte]]] =
     matches(items.BodyItem)(!_.isChunked)(req =>
       req.contentLength match {
-        case Some(n) if n > 0 => Some(Buf.ByteArray.Shared.extract(req.content))
-        case _ => None
+        case Some(0) => Some(Array.empty)
```

Comment:
Let's preallocate `Some(Array.empty[Byte])` since it's effectively immutable (there is nothing a user can do to change that array: length's is fixed and it's zero).